services/tables/src/main/java/com/linkedin/openhouse/tables/toggle/ResourceGatedByToggledOnFeatureException.java

```diff
@@ -0,0 +1,30 @@
+package com.linkedin.openhouse.tables.toggle;
+
+/**
+ * The exception thrown when the request feature of a resource is toggled on thus being gated access
+ * or modification.
+ */
+public class ResourceGatedByToggledOnFeatureException extends RuntimeException {
+  public ResourceGatedByToggledOnFeatureException(
+      String featureId, String databaseId, String tableId, Throwable throwable) {
+    this(
+        String.format(
+            "Feature [%s] has been toggled on for the [%s.%s]", featureId, databaseId, tableId),
+        throwable);
+  }
+
+  public ResourceGatedByToggledOnFeatureException(
+      String featureId, String databaseId, String tableId) {
+    this(
+        String.format(
+            "Feature [%s] has been toggled on for the [%s.%s]", featureId, databaseId, tableId));
+  }
+
+  public ResourceGatedByToggledOnFeatureException(String message, Throwable throwable) {
```

Comment:
replace this with lombok's @StandardException?

Reply:
Our Lombok version is 1.18.20; This feature is added in 1.18.22. 
OK to pursue that but since Lombok generated code get into runtime, that will involver bigger testing efforts. Shall we do this separately?